Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -0,0 +1,301 @@
+                synchronized (servers) {
+                    servers.addAll(subServers);
+                }
+            } catch (IOException e) {
+                // swallow exception and skip error subscription
+            }
+        });
+        return servers;
+    }
+
+    public static Server findServer(final String subscriptionId, final String resourceGroup, final String name) {
+        final MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        Server result = manager.servers().getByResourceGroup(resourceGroup, name);
+        return result;
+    }
+
+    public static Server create(final String subscriptionId, final String resourceGroupName, final String serverName,
+                                Region region, final ServerPropertiesForDefaultCreate properties) {
+        final MySQLManager manager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        /*Sku sku = new Sku().withName("GP_Gen5_4");*/
+        Server result = manager.servers().define(serverName).withRegion(region.name()).withExistingResourceGroup(resourceGroupName)
+                .withProperties(properties)/*.withSku(sku)*/.create();
+        return result;
+    }
+
+    public static void delete(final String subscriptionId, final String id) {
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().deleteByIds(id);
+    }
+
+    public static void start(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        Preconditions.checkArgument(
+                ServerState.fromString("Stopped").equals(currentServer.userVisibleState()) || ServerState.DISABLED.equals(currentServer.userVisibleState()),
+                "Start action is not supported for non-disabled server.");
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().inner().start(server.resourceGroupName(), server.name());
+    }
+
+    public static void restart(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
+        Preconditions.checkArgument(ServerState.READY.equals(currentServer.userVisibleState()), "Restart action is not supported for non-ready server.");
+        final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
+        mySQLManager.servers().inner().restart(server.resourceGroupName(), server.name());
+    }
+
+    public static void stop(final String subscriptionId, final Server server) {
+        Server currentServer = MySQLMvpModel.findServer(subscriptionId, server.resourceGroupName(), server.name());
```

Comment:
May move the codes for status check and get SQLManager to a utils method

Reply:
SQLManager is a resource manager. MySQL server is a instance. they are different entities.